packages/@aws-cdk/cx-api/lib/assembly/cloud-artifact.ts

```diff
@@ -0,0 +1,96 @@
+import { Artifact, ArtifactType } from '../artifacts';
+import { ERROR_METADATA_KEY, INFO_METADATA_KEY, MetadataEntry, MissingContext, WARNING_METADATA_KEY } from '../cxapi';
+import { CloudAssembly } from './cloud-assembly';
+import { Environment, ICloudArtifact, SynthesisMessage, SynthesisMessageLevel } from './cloud-assembly-api';
+
+const ENVIRONMENT_PARSER = /aws:\/\/([0-9a-z\-]+)\/([a-z\-0-9]+)/;
+
+export class CloudArtifact implements ICloudArtifact {
+  public static from(assembly: CloudAssembly, name: string, artifact: Artifact): CloudArtifact {
+    switch (artifact.type) {
+      case ArtifactType.AwsCloudFormationStack:
+        return new CloudFormationStackArtifact(assembly, name, artifact);
+
+      default:
+        throw new Error(`unsupported artifact type: ${artifact.type}`);
+    }
+  }
+
+  public readonly missing: { [key: string]: MissingContext };
+  public readonly autoDeploy: boolean;
+  public readonly messages: SynthesisMessage[];
+  public readonly environment: Environment;
+  public readonly metadata: { [path: string]: MetadataEntry[] };
+  public readonly dependsIDs: string[];
+
+  private _deps?: ICloudArtifact[]; // cache
```

Comment:
Why it called `_deps` if it actually `_cache`?

Reply:
It's a cache of deps